### PR TITLE
[CONTP-266] Add namespaceAnnotationsAsTags to operator

### PR DESCRIPTION
**Pull request by zhuminyi**
### What does this PR do?

[CONTP-266](https://datadoghq.atlassian.net/browse/CONTP-266)

### Motivation

Expose namespaceAnnotationsAsTags option

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### Checklist

- [ ] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [ ] PR has a milestone or the `qa/skip-qa` label

[CONTP-266]: https://datadoghq.atlassian.net/browse/CONTP-266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/configuration.v2alpha1.md [hunk 1]
```diff
@@ -206,6 +206,7 @@ spec:
 | global.localService.forceEnableLocalService | ForceEnableLocalService forces the creation of the internal traffic policy service to target the agent running on the local node. This parameter only applies to Kubernetes 1.21, where the feature is in alpha and is disabled by default. (On Kubernetes 1.22+, the feature entered beta and the internal traffic service is created by default, so this parameter is ignored.) Default: false |
 | global.localService.nameOverride | NameOverride defines the name of the internal traffic service to target the agent running on the local node. |
 | global.logLevel | LogLevel sets logging verbosity. This can be overridden by container. Valid log levels are: trace, debug, info, warn, error, critical, and off. Default: 'info' |
+| global.namespaceAnnotationsAsTags | Provide a mapping of Kubernetes Namespace Annotations to Datadog Tags. <KUBERNETES_LABEL>: <DATADOG_TAG_KEY> |
```
Comment (buraizu) [inline docs/configuration.v2alpha1.md:209]:
```suggestion
| global.namespaceAnnotationsAsTags | Provide a mapping of Kubernetes namespace annotations to Datadog tags. <KUBERNETES_LABEL>: <DATADOG_TAG_KEY> |
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with one minor update requested for the markdown file

**Review by wdhif — APPROVED** [review]
nit: I'm just unsure about modifying `v1alpha1` and `v1beta1` files, but @DataDog/container-ecosystems can validate this.

**Comment by celenechang** [comment]
@zhuminyi thanks for the contribution! as wassim said, no need to edit the files in `/v1alpha1` as that will be removed soon. in addition, no need to update `bundle/manifests`; that will be taken care of during the release process. do you mind removing those changes?

also, do you mind fixing the issues in the validation job?

**Review by celenechang — APPROVED** [review]

**Pull request merged**